src/modules/SignIn/components/SignIn/components/FormSignIn/FormSignIn.jsx

```diff
@@ -46,35 +49,80 @@ export default class FormSignIn extends Component {
     onSubmit(event) {
         event.preventDefault();
         const { formData } = this.state;
-        const validateResult = validSignInForm(formData);
-
-        if (validateResult.status) {
-            API.signIn(formData)
-                .then((response) => {
-                    if (response.status === signInSuccessCode) {
-                        setTokens(response.data.access, response.data.refresh);
-                        window.location.href = "/dashboard";
-                    }
-                })
-                .catch((error) => {
-                    if (error.response.status === signInErrorCode) {
-                        this.setState(() => ({
-                            errorMsg: signInErrorMsg,
-                        }));
+
+        API.signIn(formData)
+            .then((response) => {
+                if (response.status === 200) {
+                    setTokens(response.data.access, response.data.refresh);
+                    window.location.href = "/dashboard";
+                }
+            })
+            .catch((error) => {
+                if (error.response.status === 400) {
+                    let incorrectStatus = false;
+
+                    if (Object.keys(error.response.data).includes("non_field_errors")) {
+                        incorrectStatus = true;
                     }
-                });
-        } else {
-            Object.keys(validateResult.invalidFields).forEach((field) => {
-                const fieldStatus = validateResult.invalidFields[field];
-                this.setState((prevState) => ({
-                    isValid: Object.assign(prevState.isValid, { [field]: fieldStatus }),
-                }));
+
+                    this.setState(() => ({
+                        errorMsg: signInErrorMsg,
+                        incorrectStatus,
+                    }));
+
+                    Object.keys(this.fieldRefs).forEach((ref) => {
+                        this.fieldRefs[ref].current.forceUpdate();
+                    });
+                }
             });
+    }
+
+    usernameValidor(value) {
+        let usernameValid = true;
+        if (!value) {
+            usernameValid = false;
+        }
+
+        this.setState((prevState) => ({
+            buttonStates: {
+                ...prevState.buttonStates,
+                username: usernameValid,
+            },
+        }));
+
+        return usernameValid;
+    }
+
+    passwordValidor(value) {
+        let usernameValid = true;
+        if (!value) {
```

Comment:
You can replace a conditional expression with `const usernameValid = Boolean(value);`.